stablehlo/reference/Ops.cpp

```diff
@@ -47,5 +47,37 @@ Tensor eval(SineOp op, const Tensor &operand) {
   return result;
 }
 
+namespace {
+
+// Appies permutation `perm` to an array `array` where perm[i] indicates the
+// location where the current array[i] goes.
+void applyInPlacePermutation(std::vector<int64_t> &array,
+                             const std::vector<int64_t> &perm) {
```

Comment:
Is there a strong reason to apply the permutation in place? Functional style would be much easier to understand, e.g.:

```
llvm::SmallVector<int64_t> permute(ArrayRef<int64_t> array, DenseIntElementsAttr permutation) {
  llvm::SmallVector<int64_t> result(array.size());
  for (auto i = 0; i < result.size(); ++i) {
    result[permutation.getValues<int64_t>()[i]] = array[i];
  }
  return result;
}
```

Reply:
The in-place permute is not needed. Replaced it with a simpler implementation.